src/main/java/game/MoveList.java

```diff
@@ -56,18 +59,31 @@ public void sort() {
         Arrays.sort(buffer, 0, size, null);
     }
 
+    public boolean anyMatch(Predicate<Move> predicate) {
+        for (int i = 0; i < size; i++) {
+            if (predicate.test(buffer[i]))
+                return true;
+        }
+        return false;
+    }
+
     public void retain(Predicate<Move> predicate) {
+        // Keep only the moves where the predicate returns true. Does not
+        // preserve the order of moves.
         int i = 0;
         while (i < size) {
             if (predicate.test(buffer[i])) {
-                i += 1;
+                i++;
             } else {
                 swapRemove(i);
             }
         }
     }
 
     private void swapRemove(int i) {
+        // Removes a move by swapping the last move into its place.
```

Comment:
This first line could be put in javadoc style:

```
/**
 * Removes a move by swapping the last move into its place
 */
     private void swapRemove(int i) {
```

Reply:
Python hacker spotted.